framework/PluginBaseClasses/PluginBase.py

```diff
@@ -16,29 +16,25 @@
 
 @author: alfoa
 """
-from __future__ import division, print_function , unicode_literals, absolute_import
-
-#External Modules---------------------------------------------------------------
-#External Modules End-----------------------------------------------------------
-
-#Internal Modules---------------------------------------------------------------
-#Internal Modules End-----------------------------------------------------------
 
 class PluginBase(object):
   """
     This class represents an abstract class each specialized plugin class should inherit from
   """
   # List containing the methods that need to be checked in order to assess the
   # validity of a certain plugin. This list needs to be populated by the derived class
+  _interfaceFactory = None # set in each class, is the factory for the related interface module
```

Comment:
Can you make a method to force the developer to set ```_interfaceFactory```?

Reply:
If the core RAVEN developer does not add this, then it will not run. I think that is sufficient clue, again for a core developer.